src/session/content/message_row/sticker.rs

```diff
@@ -122,7 +96,37 @@ impl MessageSticker {
 
     fn load_sticker(&self, path: &str) {
         let self_ = imp::MessageSticker::from_instance(self);
-        let media = gtk::MediaFile::for_filename(path);
-        self_.picture.set_paintable(Some(&media));
+        let file = gio::File::for_path(path);
+        let future = clone!(@weak self_.paintable as paintable => async move {
+            match file.load_bytes_async_future().await {
+                Ok((bytes, _)) => {
+                    let image = webp::Decoder::new(&bytes)
+                        .decode()
+                        .unwrap()
+                        .to_image()
+                        .to_rgba8();
+
+                    let samples = image.into_flat_samples();
+
+                    let (stride, width, height) = samples.extents();
+                    let gtk_stride = stride * width;
+
+                    let bytes = glib::Bytes::from_owned(samples.samples);
```

Comment:
`samples.samples` looks a bit weird. I suggest renaming `samples` to `flat_samples`.